src/internal/m365/collection/site/restore.go

```diff
@@ -156,8 +158,23 @@ func restoreListItem(
 
 	newName := fmt.Sprintf("%s_%s", destName, listName)
 
+	storedList, err := api.BytesToListable(bytes)
+	if err != nil {
+		return dii, clues.WrapWC(ctx, err, "generating list from stored bytes")
+	}
+
+	// newName is of format: destinationName_listID
```

Comment:
Should we add a unit test which tests for expected list display name? Maybe to `TestListCollection_Restore`?